com.unity.multiplayer.mlapi/Tests/Runtime/RpcQueueTests.cs

```diff
@@ -78,33 +47,158 @@ public IEnumerator RpcQueueUnitTest()
                 yield return new WaitForSeconds(0.02f);
 
                 testsAreComplete = rpcPipelineTestComponent.IsTestComplete();
-                exceededMaximumStageIterations = rpcPipelineTestComponent.ExceededMaxIterations();
+                Assert.IsFalse(rpcPipelineTestComponent.ExceededMaxIterations());
             }
+            var testsAreValidated = rpcPipelineTestComponent.ValidateUpdateStages();
 
-            if (!exceededMaximumStageIterations)
-            {
-                testsAreValidated = rpcPipelineTestComponent.ValidateUpdateStages();
-            }
+            //Stop testing
+            rpcPipelineTestComponent.EnableTesting = false;
+
+            Debug.Log($"Exiting status => {nameof(testsAreComplete)}: {testsAreComplete} - {nameof(testsAreValidated)}: {testsAreValidated} -{nameof(exceededMaximumStageIterations)}: {exceededMaximumStageIterations}");
+
+            Assert.IsTrue(testsAreComplete && testsAreValidated);
+
+            //Disable this so it isn't running any longer.
+            rpcPipelineTestComponent.gameObject.SetActive(false);
+
+            yield return null;
+#else
+            yield return null;
+#endif
+        }
 
-            //Stop pinging
-            rpcPipelineTestComponent.PingSelfEnabled = false;
-            Debug.Log("RPC Queue Testing completed.");
+        /// <summary>
+        /// This tests the RPC Queue outbound and inbound buffer capabilities.
+        /// It will send
+        /// </summary>
+        /// <returns>IEnumerator</returns>
+        [UnityTest]
+        public IEnumerator BufferDataValidation()
+        {
```

Comment:
```suggestion
        [Test]
        public void BufferDataValidation()
        {
```

if you're not yielding anything, you could turn this into a plain test (that'd also work fine too!)

Reply:
It is yielding ? https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/pull/577/files#diff-f3da7f80c131d4b8fbc19fc6f4873635cccc9a43afcc863986b6018c5f5d9d4fR95